NugetLightRequest.cs

```diff
@@ -1909,5 +1929,134 @@ private IEnumerable<PackageItem> SearchForPackages(PackageSource source, string
                 return Enumerable.Empty<PackageItem>();
             }
         }
+
+        internal NetworkCredential GetCredsFromCredProvider(string query, NuGetRequest request, bool isRetry=false)
+        {
+            string username = "";
+            string password = "";
+            
+            // Find credential provider
+            // Option 1. Use env var 'NUGET_PLUGIN_PATHS' to find credential provider
+            // Note: unix can only use option 1
+            string credProviderPath = "";
+            string NugetPluginPath = Environment.GetEnvironmentVariable("NUGET_PLUGIN_PATHS");
+            string NuGetPluginsPathNetCore = Environment.ExpandEnvironmentVariables("%UserProfile%/.nuget/plugins/netcore/CredentialProvider.Microsoft/CredentialProvider.Microsoft.exe");
+            string NuGetPluginsPathNetFramework = Environment.ExpandEnvironmentVariables("%UserProfile%/.nuget/plugins/netfx/CredentialProvider.Microsoft/CredentialProvider.Microsoft.exe");
+            if (!NugetPluginPath.IsNullOrEmpty())
+            {
+                credProviderPath = NugetPluginPath;
+            }
+            else if (File.Exists(NuGetPluginsPathNetCore)) 
+            {
+                credProviderPath = NuGetPluginsPathNetCore;
+            }
+            else if (File.Exists(NuGetPluginsPathNetFramework))
+            {
+                credProviderPath = NuGetPluginsPathNetFramework;
+            }
```

Comment:
will it be a problem .net core vs full .net? any official place we can pick instead of searching everywhere?